galsim/utilities.py

```diff
@@ -1331,3 +1331,46 @@ def rand_with_replacement(n, n_choices, rng, weight=None, _n_rng_calls=False):
         return index, n_rng_calls
     else:
         return index
+
+
+def check_share_file(filename, subdir):
+    """Find SED or Bandpass file, possibly adding share dir or raising deprecation warning if old
+    share dir was specified.
+    """
+    from .deprecated import depr
+    import os
+
+    if os.path.isfile(filename):
+        return True, filename
+
+    new_filename = os.path.join(galsim.meta_data.share_dir, subdir, filename)
+    if os.path.isfile(new_filename):
+        return True, new_filename
+
+    dirname, basename = os.path.split(filename)
+    new_filename = os.path.join(dirname, subdir, basename)
+    if os.path.isfile(new_filename):
+        depr("Filename os.path.join(galsim.meta_data.share_dir, {0})".format(basename),
+             1.5,
+             "os.path.join(galsim.meta_data.share_dir, '{0}', {1})".format(subdir, basename))
+        return True, new_filename
+
+    return False, ''
+
+
+# http://stackoverflow.com/a/6849299
+class lazy_property(object):
+    """
+    meant to be used for lazy evaluation of an object attribute.
+    property should represent non-mutable data, as it replaces itself.
+    """
+    def __init__(self, fget):
+        self.fget = fget
+        self.func_name = fget.__name__
+
+    def __get__(self, obj, cls):
+        if obj is None:
+            return None
```

Comment:
I think `obj=None` occurs if you try to access the property from the class instead of from an instance of the class.  (See https://docs.python.org/3/reference/datamodel.html#invoking-descriptors).  I kept it mostly b/c it was in the SO answer and in the pure python description of a `@property`: https://docs.python.org/3/howto/descriptor.html#properties.

So I guess we need to decide what ought to happen if a user tries to access, e.g., `galsim.Gaussian.noise`...  I don't have a strong opinion on this, but returning `None` seems reasonable (though this would also be the return value without this branch I think).